modules/gdscript/gdscript_compiler.cpp

```diff
@@ -920,7 +924,7 @@ GDScriptCodeGenerator::Address GDScriptCompiler::_parse_expression(CodeGen &code
 								StringName var_name = identifier->name;
 								if (_is_class_member_property(codegen, var_name)) {
 									assign_class_member_property = var_name;
-								} else if (!codegen.locals.has(var_name) && codegen.script->member_indices.has(var_name)) {
+								} else if (!_is_local_or_parameter(codegen, var_name) && codegen.script->member_indices.has(var_name)) {
```

Comment:
I see the other similar change (line 1134L/1138R) is in _"Regular assignment"_ case so I suppose that's the one which actually fixes the example you've showcased (CMIIW). What case does the change in here fix?

Reply:
Hm... as it is visible that this is about an assignment a to subscript so `a.b = 3`, and this part in particular is about deciding the origin of the base - `a`. It makes the same mistake as bellow and sets bunch of stuff like `is_member_property` to true for actually shadowed by parameter hypothetical `a`, but for some reason it still works afterwards as expected, at least it is hard for me to create a clear test to show the difference. Will investigate more.

Reply:
Added example with broken subscript assignment to the test.

Reply:
Yeah, I was wondering whether a test for it could be added too. Thanks for doing so!